src/app/modules/shared-components/components/page-navigator/page-navigator.component.ts

```diff
@@ -21,9 +21,7 @@ export class PageNavigatorComponent {
 
   isCurrentGroupWatched$ = this.userSessionService.session$.pipe(
     filter(isNotNullOrUndefined),
-    map(data => data.watchedGroup),
-    filter(isNotNullOrUndefined),
-    map(watchedGroup => this.groupId === watchedGroup.id)
+    map(data => data?.watchedGroup?.id === this.groupId)
```

Comment:
If data cannot be null (with your previous line), there is no reason for the first `?` I assume.
And a check that both are not undefined would be cleaner.

So I would expect something like
```
map(data => this.groupId !== undefined && data.watchedGroup?.id === this.groupId)
```

Reply:
Fixed